### ISSUE TITLE
Unpickling of previously-created traits fails due to `__newobj__` move.

### ISSUE THREAD

**Issue by mdickinson** [enthought/traits#781]
A first test of unpickling of previously-created cTrait instances failed for me.

The following is a pickle of a cTrait created on Python 2.7 / Traits 5.2.0, using `pickle.dumps()`. (The Python 2.7 bit isn't particularly important. The Traits 5.2.0 bit is.)

```
>>> import pickle, traits.api
>>> pickle.dumps(traits.api.BaseFloat(5.3).as_ctrait())
"ctraits.traits\n__newobj__\np0\n(ctraits.traits\nCTrait\np1\nI0\ntp2\nRp3\n(I0\nI0\nI5\nNI14\nI-1\nI0\nF5.3\nI0\nNNI4\nNccopy_reg\n_reconstructor\np4\n(ctraits.trait_types\nBaseFloat\np5\nc__builtin__\nobject\np6\nNtp7\nRp8\n(dp9\nS'default_value_type'\np10\nI0\nsS'_metadata'\np11\n(dp12\nS'type'\np13\nS'trait'\np14\nssS'default_value'\np15\nF5.3\nsb(dp16\ng13\ng14\nstp17\nb."
```

Attempting to unpickle this with Traits master (on Python 3.8) gives:
```
AttributeError: Can't get attribute '__newobj__' on <module 'traits.traits' from '/Users/mdickinson/Enthought/ETS/traits/traits/traits.py'>
```

**Comment by mdickinson** [comment]
I think the quick fix is to add back a `__newobj__` function to `traits.traits`, with a comment explaining why it's there (and a regression test).

It's not worth importing the existing `__newobj__` from other modules; that just adds unnecessary inter-module dependencies.

**Comment by mdickinson** [comment]
This was fixed in #787.
